src/sbt-test/caching/unzip/build.sbt

```diff
@@ -0,0 +1,98 @@
+lazy val tempUnzipDir = IO.createTemporaryDirectory
+
+lazy val root = (project in file(".")).
+  settings(
+    version := "0.1",
+    scalaVersion := "2.11.12",
+    libraryDependencies += "commons-io" % "commons-io" % "2.4",
+    libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.1" % "test",
+    libraryDependencies += "ch.qos.logback" % "logback-classic" % "0.9.29" % "runtime",
+    assemblyShadeRules in assembly := Seq(
+      ShadeRule
+        .rename("org.apache.commons.io.**" -> "shadeio.@1")
+        .inLibrary("commons-io" % "commons-io" % "2.4")
+        .inProject
+    ),
+    assemblyOption in assembly ~= {
+      _.withCacheOutput(true)
+        .withCacheUnzip(true)
+        .withAssemblyUnzipDirectory(tempUnzipDir)
+    },
+    assemblyJarName in assembly := "foo.jar",
+    TaskKey[Seq[File]]("genresource") := {
+      val dirs = (unmanagedResourceDirectories in Compile).value
+      val file = dirs.head / "foo.txt"
+      IO.write(file, "bye")
+      Seq(file)
+    },
+    TaskKey[Seq[File]]("genresource2") := {
+      val dirs = (unmanagedResourceDirectories in Compile).value
+      val file = dirs.head / "bar.txt"
+      IO.write(file, "bye")
+      Seq(file)
+    },
+    TaskKey[Unit]("check") := {
+      val process = sys.process.Process("java", Seq("-jar", (crossTarget.value / "foo.jar").toString))
+      val out = (process!!)
+      if (out.trim != "hello") sys.error("unexpected output: " + out)
+      ()
+    },
+    TaskKey[Unit]("checkfoo") := {
+      val process = sys.process.Process("java", Seq("-jar", (crossTarget.value / "foo.jar").toString))
+      val out = (process!!)
+      if (out.trim != "foo.txt") sys.error("unexpected output: " + out)
+      ()
+    },
+    TaskKey[Unit]("checkhash") := {
+      val s = streams.value
+      val jarHash = crossTarget.value / "jarHash.txt"
+      val hash = getHashString(crossTarget.value / "foo.jar")
+      if ( jarHash.exists )
+      {
+        val prevHash = IO.read(jarHash)
+        s.log.info( "Checking hash: " + hash + ", " + prevHash )
+        assert( hash == prevHash )
+      }
+      IO.write( jarHash, hash )
+      ()
+    },
+    TaskKey[Unit]("checkunzip") := {
```

Comment:
@eed3si9n I think this covers the unit tests I was thinking of, do you have any more ideas?